apps/els_lsp/src/els_fungraph.erl

```diff
@@ -0,0 +1,43 @@
+-module(els_fungraph).
```

Comment:
I wonder if we could avoid rolling out our own traversal by using one of the `digraph_util` helpers or similar.

Reply:
Let's eventually address this in a follow up PR.

Reply:
> I wonder if we could avoid rolling out our own traversal by using one of the `digraph_util` helpers or similar.

From what I was able to conceive so far while working on the codebase, this would probably make perfect sense if we had a complete digraph at hand containing all the relationships (e.g. "includes", "depends on") between different modules and headers. Though this digraph would have to be built and maintained during indexing I believe. The "shared" nature of the digraph internal representation (aka 3 ETS tables) would even provide extra benefit for performance.

Reply:
> Before doing squashing I wondered if I should also attempt to express [this routine](https://github.com/erlang-ls/erlang_ls/blob/1f6f9edd2089c86d54729abe4074c1d52ea71263/apps/els_lsp/src/els_code_navigation.erl#L139-L151) with fungraph traversal. Or maybe it's followup PRs' material?

I think that's totally fine not to include here.